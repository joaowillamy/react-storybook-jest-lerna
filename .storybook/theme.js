import { create } from "@storybook/theming/create";

export default create({
  base: "light",

  colorPrimary: "#FDCB13",
  colorSecondary: "#18ACC4",

  // UI
  appBg: "white",
  appContentBg: "#F2F3F4",
  appBorderColor: "#F2F3F4",
  appBorderRadius: 4,

  // Typography
  fontBase: '"Montserrat", sans-serif',
  fontCode: "monospace",

  brandTitle: "Quero front test"
  // brandUrl: 'http://resolvvi.com.br/',
  // brandImage: ''
});
