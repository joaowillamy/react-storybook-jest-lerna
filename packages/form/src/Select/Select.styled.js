import styled from "styled-components";

export const Select = styled.div`
  div {
    background-color: #fff;
    border-radius: 5px;
  }
`;
