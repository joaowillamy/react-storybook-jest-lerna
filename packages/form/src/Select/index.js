import React from "react";
import ReactSelect from "react-dropdown-select";
import PropTypes from "prop-types";

import * as S from "./Select.styled";

const Select = props => {
  return (
    <S.Select>
      <ReactSelect searchable={false} placeholder="Selecione..." {...props} />
    </S.Select>
  );
};

Select.propTypes = {
  labelField: PropTypes.string,
  valueField: PropTypes.string,
  options: PropTypes.array,
  onChange: PropTypes.func
};

export default Select;
