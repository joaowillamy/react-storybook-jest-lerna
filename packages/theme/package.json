{
  "name": "@will-ui-kit/theme",
  "version": "1.0.2",
  "main": "lib/index.js",
  "module": "src/index.js"
}
