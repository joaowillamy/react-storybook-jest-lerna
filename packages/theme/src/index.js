export { default as ThemeProvider } from "./ThemeProvider";
export { default as DEFAULT_THEME } from "./theme";
export { GlobalStyles } from "./GlobalStyles";
