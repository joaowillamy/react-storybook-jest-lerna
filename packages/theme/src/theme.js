const BASE = 3;

const borders = {
  sm: color => `1px solid ${color || "#00000029"}`,
  md: color => `3px solid ${color || "#00000029"}`
};

const colors = {
  bluePrimary: "#18ACC4",
  blueSecondary: "#007A8D",
  yellowPrimary: "#FDCB13",
  yellowSecondary: "#DE9E1F",
  green: "#0FA866",
  black: "#1F2D30",
  grey: "#FBFBFB",
  Overlay: "#1F2D30e0" // com 88% de opacidade.
};

const fonts = {
  Montserrat: "Montserrat"
};

const fontSizes = {
  xs: "12px",
  sm: "14px",
  md: "16px",
  lg: "18px",
  xl: "22px",
  xxl: "26px",
  xxxl: "36px"
};

const fontWeights = {
  thin: 100,
  extralight: 200,
  light: 300,
  regular: 400,
  medium: 500,
  semibold: 600,
  bold: 700,
  extrabold: 800,
  black: 900
};

const iconSizes = {
  sm: "12px",
  md: "16px",
  lg: "26px"
};

const shadows = {
  sm: color => `0 1px 3px ${color || "#00000029"}`,
  md: color => `0 3px 4px ${color || "#00000029"}`,
  lg: color => `0 3px 6px ${color || "#00000029"}`
};

const space = {
  xs: `${BASE}px`,
  sm: `${BASE * 2}px`,
  md: `${BASE * 3}px`,
  lg: `${BASE * 4}px`,
  xl: `${BASE * 8}px`,
  xxl: `${BASE * 10}px`,
  xxxl: `${BASE * 12}px`
};

const DEFAULT_THEME = {
  borders,
  colors,
  fonts,
  fontSizes,
  fontWeights,
  iconSizes,
  shadows,
  space
};

export default DEFAULT_THEME;
