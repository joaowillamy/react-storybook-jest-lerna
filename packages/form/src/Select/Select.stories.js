import React from "react";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import Select from ".";

export default {
  component: Select,
  title: "form|Select"
};

export const select = () => {
  return (
    <ThemeProvider>
      <Select
        labelField={"name"}
        // eslint-disable-next-line no-console
        onChange={values => console.log(values)}
        options={[
          {
            id: "ea304469-3d25-4b33-967b-9740ab7e910f",
            name: "Madge Casper"
          },
          {
            id: "ea30446asd9-3d25-4b33-967b-9740ab7e910f",
            name: "Joao"
          }
        ]}
        valueField={"id"}
      />
    </ThemeProvider>
  );
};
