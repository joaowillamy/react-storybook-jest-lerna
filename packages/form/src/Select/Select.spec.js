import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import Select from ".";

describe("Select", () => {
  test("Select", () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <Select
            labelField={"name"}
            options={[
              {
                id: "ea304469-3d25-4b33-967b-9740ab7e910f",
                name: "Madge Casper"
              },
              {
                id: "ea30446asd9-3d25-4b33-967b-9740ab7e910f",
                name: "Joao"
              }
            ]}
            valueField={"id"}
          />
        </ThemeProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
