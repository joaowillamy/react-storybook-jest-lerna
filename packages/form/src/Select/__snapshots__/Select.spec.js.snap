// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Select 1`] = `
.c0 div {
  background-color: #fff;
  border-radius: 5px;
}

<div
  className="c0"
>
  <div>
    <div
      className="react-dropdown-select undefined css-wmw4vi-ReactDropdownSelect e1gzf2xs0"
      color="#0074D9"
      direction="ltr"
      disabled={false}
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex="0"
    >
      <div
        className="react-dropdown-select-content react-dropdown-select-type-single css-v1jrxw-ContentComponent e1gn6jc30"
        onClick={[Function]}
      >
        <input
          className="react-dropdown-select-input css-1b6aba5-InputComponent e11wid6y0"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          placeholder="Selecione..."
          readOnly={true}
          size={12}
          tabIndex="-1"
          value=""
        />
      </div>
      <div
        className="react-dropdown-select-dropdown-handle css-1yc4zyy-DropdownHandleComponent e1vudypg0"
        color="#0074D9"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        rotate={1}
        tabIndex="-1"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 40 40"
        >
          <path
            d="M31 26.4q0 .3-.2.5l-1.1 1.2q-.3.2-.6.2t-.5-.2l-8.7-8.8-8.8 8.8q-.2.2-.5.2t-.5-.2l-1.2-1.2q-.2-.2-.2-.5t.2-.5l10.4-10.4q.3-.2.6-.2t.5.2l10.4 10.4q.2.2.2.5z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
